### ISSUE TITLE
[FEATURE] save optimizer state_dict and scheduler_dict

### ISSUE THREAD

**Issue by Quetzalcohuatl** [h2oai/h2o-llmstudio#571]
### 🚀 Feature

We already save checkpoint.pth , optionally save optimizer state dict and scheduler state dict so you can reload from checkpoint and continue training in case training run breaks

**Comment by psinger** [comment]
Yes resuming would be a great feature already tracked here: https://github.com/h2oai/h2o-llmstudio/issues/431

Closing this as duplicate.
